core/templates/dev/head/domain/skill/SkillObjectFactory.js

```diff
@@ -23,16 +23,14 @@ oppia.factory('SkillObjectFactory', [
       ConceptCardObjectFactory, MisconceptionObjectFactory, ValidatorsService) {
     var Skill = function(
         id, description, misconceptions, conceptCard, languageCode, version,
-        nextMisconceptionId, supersedingSkillId, allQuestionsMerged) {
+        nextMisconceptionId) {
       this._id = id;
       this._description = description;
       this._misconceptions = misconceptions;
       this._conceptCard = conceptCard;
       this._languageCode = languageCode;
       this._version = version;
       this._nextMisconceptionId = nextMisconceptionId;
-      this._supersedingSkillId = supersedingSkillId;
```

Comment:
Hang on, I think there may have been a miscommunication here. If the two extra fields are part of the definition of the backend object, it's fine to keep them in the frontend object as well. What I was suggesting we remove are just the **functions** that operate on those fields, if nothing else in the frontend uses them.

I.e. it is fine (and probably a good idea) to use the argument "let's keep parity with the backend" for domain objects, but that argument doesn't extend to methods/functions on those objects.